### PR TITLE
Fix critical_processes_list in devices.py

### PR DESCRIPTION
**Pull request by bingwang-ms**
### Description of PR

Summary:
Fixes # (issue)
Fix issue in 
The critical_process_list got in critical_process_status method is malformated. This commit fix the issue.
### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
The critical_process_status in sanity_check gets critital list with command **cat /etc/supervisor/critical_processes**. However, the output of the command is like following:
```
root:/# cat /etc/supervisor/critical_processes
program:ledd
program:xcvrd
program:psud
```
There is a leading *program* in each line, and as a result, the following operation
```
if pname in critical_process_list:
                    result['running_critical_process'].append(pname)
```
will always get an empty list, and critital processes are not checked correctly. 
```
 "processes_status": {
            "lldp": {
                "status": true, 
                "exited_critical_process": [], 
                "running_critical_process": []
            }, 
            "pmon": {
                "status": true, 
                "exited_critical_process": [], 
                "running_critical_process": []
            }, 
            "database": {
                "status": true, 
                "exited_critical_process": [], 
                "running_critical_process": []
            }, 
            "snmp": {
                "status": true, 
                "exited_critical_process": [], 
                "running_critical_process": []
            }, 
            "bgp": {
                "status": true, 
                "exited_critical_process": [], 
                "running_critical_process": []
            }, 
            "teamd": {
                "status": true, 
                "exited_critical_process": [], 
                "running_critical_process": []
            }, 
            "syncd": {
                "status": true, 
                "exited_critical_process": [], 
                "running_critical_process": []
            }, 
            "swss": {
                "status": true, 
                "exited_critical_process": [], 
                "running_critical_process": []
            }
        }
```
#### How did you do it?
Update code in get critical_process_list.
#### How did you verify/test it?
Verify in on Arista-7260cx3, and the running_critical_process is no longer empty
```
"swss": {
                "status": true, 
                "exited_critical_process": [], 
                "running_critical_process": [
                    "buffermgrd", 
                    "intfmgrd", 
                    "nbrmgrd", 
                    "neighsyncd", 
                    "orchagent", 
                    "portmgrd", 
                    "portsyncd", 
                    "vlanmgrd", 
                    "vrfmgrd", 
                    "vxlanmgrd"
                ]
            }
```
#### Any platform specific information?
No.
#### Supported testbed topology if it's a new test case?
N/A
### Documentation 

N/A

### CODE REVIEW

File: tests/common/devices.py [hunk 1]
```diff
@@ -337,7 +337,7 @@ def critical_process_status(self, service):
         # get critical process list for the service
         output = self.command("docker exec {} bash -c '[ -f /etc/supervisor/critical_processes ] && cat /etc/supervisor/critical_processes'".format(service), module_ignore_errors=True)
         for l in output['stdout'].split():
-            critical_process_list.append(l.rstrip())
+            critical_process_list.append(l.split(':')[1].rstrip())
```
Comment (lguohan) [inline tests/common/devices.py:340]: can you check if ':' exists? if yes, then do the split, if not, then do not split. I do not know if this criticial_processes_list has been ported back to 201911 branch.

Comment (bingwang-ms) [reply tests/common/devices.py:340]: > can you check if ':' exists? if yes, then do the split, if not, then do not split. I do not know if this criticial_processes_list has been ported back to 201911 branch.

Yes. I checked containers in  SONiC.20191130.40, and the ':' exists. One of the output is below:
```
root@str-msn2700-02:/# cat /etc/supervisor/critical_processes
program:orchagent
program:portsyncd
program:neighsyncd
program:vlanmgrd
program:intfmgrd
program:portmgrd
program:buffermgrd
program:vrfmgrd
program:nbrmgrd
program:vxlanmgrd
```

Comment (lguohan) [reply tests/common/devices.py:340]: sorry, I meant the test to check if there is colon, if yes, then split, if no, the no split. this is the improve the test backward compatibility.

Comment (bingwang-ms) [reply tests/common/devices.py]: > sorry, I meant the test to check if there is colon, if yes, then split, if no, the no split. this is to improve the test backward compatibility.

Updated. Thanks!

**Pull request merged**